tests/unit/test_analytics.py

```diff
@@ -1,6 +1,40 @@
+import json
 import pytest
+import mock
 
-from dvc.analytics import Analytics
+from dvc import analytics
+from dvc.utils.compat import builtin_str, str
+
+
+@pytest.fixture
+def tmp_global_config(tmp_path):
+    with mock.patch(
+        "dvc.config.Config.get_global_config_dir", return_value=tmp_path
+    ):
+        yield
+
+
+def test_collect(tmp_global_config):
+    report = analytics.collect(return_code=0)
+
+    assert report["cmd_return_code"] == 0
+    assert report["scm_class"] in ["Git", None]
+    assert not report["cmd_class"]
+    assert type(report["is_binary"]) is bool
+    assert type(report["system_info"]) is dict
+    assert type(report["dvc_version"]) is builtin_str
+    assert type(report["user_id"]) is str
```

Comment:
Maybe use a `voluptuous` scheme to check the report? Would be more glancable.

Reply:
I would consider ordering of particular fields so that they match `analytics.collect`. Easier to read that way. Not essential though.